jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenLayerConfigurations.java

```diff
@@ -91,6 +110,73 @@ static JavaLayerConfigurations getForProject(
     return layerBuilder.build();
   }
 
+  /**
+   * Resolves the source files configuration for a War {@link MavenProject}.
+   *
+   * @param project the {@link MavenProject}
+   * @param extraDirectory path to the directory for the extra files layer
+   * @param appRoot root directory in the image where the app will be placed
+   * @return a {@link JavaLayerConfigurations} for the project
+   * @throws IOException if collecting the project files fails
+   */
+  private static JavaLayerConfigurations getForWarProject(
+      MavenProject project, Path extraDirectory, AbsoluteUnixPath appRoot) throws IOException {
+
```

Comment:
It'd be great if we could combine a lot of this logic with the Gradle side, but that can be in a followup PR.